File: python solutions/excel_sheet_column_number.py
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        count = 0
        for i in range(len(columnTitle)):
            if i == 0:
                count += (ord(columnTitle[-1-i]) - 64)
            else:
                count += (ord(columnTitle[-1-i]) - 64) * 26 ** i
        return count

File: ruby solutions/excel_sheet_column_number.rb
# @param {String} column_title
# @return {Integer}
def title_to_number(column_title)
    count = 0
    for i in 1..column_title.length do
        if i == 1
            count += (column_title[-i].ord - 64)
        else
            count += (column_title[-i].ord - 64) * 26 ** (i - 1)
        end
    end
    count
end